app/src/main/java/org/schabi/newpipe/player/mediabrowser/MediaBrowserConnector.kt

```diff
@@ -0,0 +1,738 @@
+
+            return Single.just(results)
+        } catch (error: Exception) {
+            return Single.error(error)
+        }
+    }
+
+    private fun handleSearchError(throwable: Throwable) {
+        Log.e(TAG, "Search error: $throwable")
+        disposePrepareOrPlayCommands()
+        playbackError(R.string.content_not_supported, PlaybackStateCompat.ERROR_CODE_NOT_SUPPORTED)
+    }
+
+    override fun onPrepareFromUri(
+        uri: Uri,
+        playWhenReady: Boolean,
+        extras: Bundle?
+    ) {
+        disposePrepareOrPlayCommands()
+        playbackError(R.string.content_not_supported, PlaybackStateCompat.ERROR_CODE_NOT_SUPPORTED)
+    }
+
+    override fun onCommand(
+        player: Player,
+        command: String,
+        extras: Bundle?,
+        cb: ResultReceiver?,
+    ) = false
+
+    fun onSearch(
+        query: String,
+        result: MediaBrowserServiceCompat.Result<List<MediaBrowserCompat.MediaItem>>,
+    ) {
+        result.detach()
+        searchDisposable?.dispose()
+        searchDisposable =
+            searchMusicBySongTitle(query)
+                .flatMap(::mediaItemsFromInfoItemList)
+                .subscribeOn(Schedulers.io())
+                .subscribe(
+                    { mediaItemsResult -> result.sendResult(mediaItemsResult) },
+                    { throwable -> this.handleSearchError(throwable) },
+                )
+    }
+
+    companion object {
+        private val TAG: String = MediaBrowserConnector::class.java.simpleName
+
+        private const val ID_AUTHORITY = BuildConfig.APPLICATION_ID
+        private const val ID_ROOT = "//$ID_AUTHORITY"
+        private const val ID_BOOKMARKS = "playlists"
+        private const val ID_HISTORY = "history"
+        private const val ID_INFO_ITEM = "item"
+
+        private const val ID_LOCAL = "local"
+        private const val ID_REMOTE = "remote"
+        private const val ID_URL = "url"
+        private const val ID_STREAM = "stream"
+        private const val ID_PLAYLIST = "playlist"
+        private const val ID_CHANNEL = "channel"
+
+        private fun infoItemTypeToString(type: InfoType): String {
+            return when (type) {
+                InfoType.STREAM -> ID_STREAM
+                InfoType.PLAYLIST -> ID_PLAYLIST
+                InfoType.CHANNEL -> ID_CHANNEL
+                else -> throw IllegalStateException("Unexpected value: $type")
+            }
+        }
+
+        private fun infoItemTypeFromString(type: String): InfoType {
+            return when (type) {
+                ID_STREAM -> InfoType.STREAM
+                ID_PLAYLIST -> InfoType.PLAYLIST
+                ID_CHANNEL -> InfoType.CHANNEL
+                else -> throw IllegalStateException("Unexpected value: $type")
+            }
+        }
+
+        private fun parseError(): ContentNotAvailableException {
+            return ContentNotAvailableException("Failed to parse media ID")
```

Comment:
Maybe this function could accept an argument to provide the media ID in question or a part of it to the exception message, even if errors are logged to Android's logcat.